### DISCUSSION TITLE
Use the plugin with Lazyvim

### DISCUSSION THREAD

**Discussion by cbaconnier** [adalessa/laravel.nvim#101] [Q&A] [answered]
I'm trying to configure with Lazyvim but `none-ls` fails to auto-complete `view('...` and `route('...`

I tried from a fresh Lazyvim installation.

checkhealth shows
> null-ls: require("null-ls.health").check()
>
> - Laravel_completion: cannot verify if the command is an executable.
> - Laravel_Diagnostics: cannot verify if the command is an executable.
> - Laravel_CodeAction: cannot verify if the command is an executable.

Both LSP and treesitter are running fine with no warning or errors.
commands like`:Laravel routes` are working fine
`:Laravel view-finder` works as well, but not `gf`

I tried with both: `phpactor` and `intelephense` 

My environment: 
- Same result on both (archlinux and nixos)
- Project's running with sail
- `tree-sitter-cli`, `fd` and `rg` are installed

**lua/plugins/laravel.lua** (Default config)
```lua
return {
  "adalessa/laravel.nvim",
  dependencies = {
    "nvim-telescope/telescope.nvim",
    "tpope/vim-dotenv",
    "MunifTanjim/nui.nvim",
    "nvimtools/none-ls.nvim",
  },
  cmd = { "Sail", "Artisan", "Composer", "Npm", "Yarn", "Laravel" },
  keys = {
    { "<leader>la", ":Laravel artisan<cr>" },
    { "<leader>lr", ":Laravel routes<cr>" },
    { "<leader>lm", ":Laravel related<cr>" },
  },
  event = { "VeryLazy" },
  opts = {
    features = {
      -- lsp_server = "intelephense",
      null_ls = {
        enable = true,
      },
      route_info = {
        enable = true, --- to enable the laravel.nvim virtual text
        position = "right", --- where to show the info (available options 'right', 'top')
        middlewares = true, --- wheather to show the middlewares section in the info
        method = true, --- wheather to show the method section in the info
        uri = true, --- wheather to show the uri section in the info
      },
    },
  },
  config = true,
}
```

I'm missing something? Thanks! 

_(I'm not a nvim/vim/lua guru. Thanks for your understanding :blush:  )_

**Comment by adalessa** [comment, marked as answer]
Sorry for the late response.
> :Laravel view-finder works as well, but not gf

Yes there is no integration for gf for now.

>I'm trying to configure with Lazyvim but none-ls fails to auto-complete view('... and route('...

On regards this, in the soon to be release version, available in development I moved away of none-ls to not make people install another dependency, and to make it work in more configuration to work with cmp.
I suggest switch to the branch, there are some breaking changes needing a new dependency, but it comes with more improvements.

**Reply by cbaconnier** [reply]
Thanks ! I will retry with the dev branch, hopefully this week. 
Also thank you very much for what you're offering for the Laravel / vim community :heart:

**Reply by adalessa** [reply]
Great, hope for the best, I have merged and tagged as version 3.0 of the plugin in case you haven't done it yet.
Thanks for the kind words, I enjoy working on it and making it better.

**Reply by cbaconnier** [reply]
@adalessa Since then I have switched from Lazyvim to NvChad. I was still no able to have autocompletion working, but I finally could make  it work by adding the `laravel` source to cmp

```lua
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      opts.sources = vim.list_extend(opts.sources or {}, {
        { name = "laravel" },
      })
    end,
  },
```
